### PR TITLE
Update to Latest Binaryen

### PR DESCRIPTION
**Pull request by MaxGraey**
* update to `binaryen@51.0.0-nightly.20180924`
* remove import calls

**FIXME**
- [ ] in some test global `$HEAP_BASE` constant not created

### CODE REVIEW

**Comment by dcodeIO** [comment]
> in some test global $HEAP_BASE constant not created

From looking over it there seems to be a HEAP_BASE each time there was one before. Which test doesn't have it?

**Comment by MaxGraey** [comment]
`compiler/std/gc-array.ts`
Fatal: Module::getGlobal: $HEAP_BASE does not exist

**Comment by dcodeIO** [comment]
Didn't that one update with `--create`? Asking because I can't find it in the changeset

**Comment by MaxGraey** [comment]
I forced run all tests with `--create` flag:
```
npm run test:parser -- --create
npm run test:compiler -- --create
```

File: src/compiler.ts [hunk 1]
```diff
@@ -5733,7 +5733,6 @@ export class Compiler extends DiagnosticEmitter {
 
     // otherwise just call through
     this.currentType = returnType;
-    if (isCallImport) return module.createCallImport(instance.internalName, operands, returnType.toNativeType());
     var ret = module.createCall(instance.internalName, operands, returnType.toNativeType());
```
[the remark this answers is not in the document]
Comment (MaxGraey) [reply src/compiler.ts]: Done!

**Comment by MaxGraey** [comment]
That happened in all tests which import `itcm` with combination of `arena` allocator:
```ts
import "allocator/arena";
import "collector/itcm";
```
Without this tests all works perfectly

**Comment by MaxGraey** [comment]
I found how this fix. But this looks like workaround:
in `std/itcm.ts` added on top current line:
```ts
var _HEAP_BASE: usize = HEAP_BASE;
```
And replace `HEAP_BASE` to `_HEAP_BASE` in other places

**Comment by MaxGraey** [comment]
Close to favour #288

**Pull request closed without merging**